src/node/orb.yml

```diff
@@ -0,0 +1,194 @@
+          key: node-deps-{{ .Branch }}-{{ checksum "<< parameters.cache-key >>" }}
+          paths:
+            - << parameters.dir >>
+  with_splitting:
+    description: |
+      Run a testing command (`test-command`) with CircleCI's built-in test splitting feature enabled. This requires
+      that the `parallelism` key be set to a value greater than 1.
+
+      `test-command` should be defined as the command used for initiating testing. The environment variable $TESTFILES
+      is defined prior to this command being called as the group of tests to be run in a given container after running
+      CircleCI's test-splitting logic.
+
+      This command requires one of `test-file`, `test-path`, or `glob-path` parameters that point
+      to a text file with test filenames, a path to test files, or one or more patterns for globbing of
+      test files, respectively.
+
+      CircleCI will then seek historical test timing data in the directory specified by the previously
+      used `store_test_results` step as a basis for splitting tests. If you do not use `store_test_results`,
+      there will be no timing data available for splitting your tests.
+
+      Tests are split by `timings` by default, and can be split based on filesize by setting the `split-by`
+      parameter to `filesize`.
+
+      The CLI expectes both filenames and classnames to be present in the timing data produced by the timing
+      suite. By default, splitting defaults to `filename`. You can specify `classnames` via the `timings-type`
+      parameter.
+
+      For more information on running tests in parallel, please visit
+      https://circleci.com/docs/2.0/parallelism-faster-jobs.
+    parameters:
+      steps:
+        type: steps
+      test-file:
+        type: string
+        default: ""
+      test-path:
+        type: string
+        default: ""
+      glob-path:
+        type: string
+        default: ""
+      split-by:
+        type: string
+        default: "timings"
+      timings-type:
+        type: string
+        default: "filename"
+    steps:
+      - when:
+          condition: << parameters.test-file >>
+          steps:
+            - run:
+                name: Split files by test file, assign to $TESTFILES
+                command: |
+                  echo 'export TESTFILES=$(circleci tests split << parameters.test-file >> | circleci tests split --split-by="<< parameters.split-by >>" --timings-type="<< parameters.timings-type >>")' >> $BASH_ENV
+                  source $BASH_ENV
+      - when:
+          condition: << parameters.test-path >>
+          steps:
+            - run:
+                name: Split files by directory, assign to $TESTFILES
+                command: |
+                  echo 'export TESTFILES=$(circleci tests split < << parameters.test-path >> | circleci tests split --split-by="<< parameters.split-by >>" --timings-type="<< parameters.timings-type >>")' >> $BASH_ENV
+                  source $BASH_ENV
```

Comment:
as far as I know, `source` like this will have no lasting effect after the command has completed, right?

Reply:
Although no impact in this step, every subsequent step should source $BASH_ENV before handing over control to the run command.

Reply:
Ah, I wasn't aware of that--was going off of our [public documentation](https://circleci.com/docs/2.0/env-vars/#setting-an-environment-variable-in-a-shell-command), will have that changed to reflect this.

Reply:
Correction to the above, `source $BASH_ENV` works as intended--it will set environment variables for use in future steps. Documentation in the previous comment